util/file.go

```diff
@@ -189,10 +189,13 @@ func CopyFolderContentsWithFilter(source string, destination string, filter func
 			if err := CopyFile(file, dest); err != nil {
 				return err
 			}
+			if err := manifest.AddFile(dest); err != nil {
+				return err
+			}
 		}
 	}
 
-	return nil
+	return manifest.Close()
```

Comment:
Should this be in a `defer` to ensure the file is closed even if we hit an error part way through?